### DISCUSSION TITLE
Vite doesn't pick up tailwind config changes. What am I doing wrong here?

### DISCUSSION THREAD

**Discussion by laygir** [tailwindlabs/tailwindcss#10865] [Help] [answered]
Hello 👋

I am struggling to figure out why vite doesn't pick up the changes I do in `tailwind.config`. There is hardcore caching clearly and I can't seem to find what am I doing wrong. Any opinions?

Here is the reproduction case https://github.com/laygir/vue-vite-tailwind

And this is a screen recording of the frustration.

https://user-images.githubusercontent.com/772730/227648443-57410cc6-921e-417e-8116-60911116ae6a.mp4

*Console in the browser logs the test color from the result of calling `resolveConfig(tailwindConfig)`

**Comment by wongjn** [comment, marked as answer]
From reading the Vite documentation[^1][^2], it seems like they expect that you'd only be converting third-party dependency modules and not your own code. I assume that this is because they assume that you'd write your own code in ESM with Vite. Unfortunately you can't do that with the Tailwind configuration in stable until #10785 is in. You could try:

- Using ESM Tailwind configuration to avoid the prebundling by testing with `insiders`, `npm install tailwindcss@insiders`.
- Clear Vite's pre-bundle cache, by passing `--force` to `vite`, e.g. as per your project, `npm run dev -- --force`. Note, you were running `npm run dev --force` (note the missing `--`) which means you were passing the `--force` flag to `npm run`, not the `vite` command that the `dev` task runs.

[^1]: https://vitejs.dev/guide/dep-pre-bundling.html
[^2]: https://vitejs.dev/config/dep-optimization-options.html

**Reply by laygir** [reply]
Thank you! 🙏 indeed once the force flag is passed properly it does what it promises.
I was expecting that additional `--` would bite me one day, keep forgetting that 🤦🏻‍♂️

–
I also did try `tailwindcss@insiders` but getting an error there when calling `resolveConfig()`.
But no worries, I'll continue with `-- --force` meanwhile.
<img width="747" alt="Screenshot 2023-03-25 at 00 35 24" src="https://user-images.githubusercontent.com/772730/227663648-caa38845-047c-4c00-8b9c-d2f4dbbf700a.png">
